doc/dev/decisionrecords/Codestyle.md

```diff
@@ -158,6 +158,21 @@ What to put in Javadoc:
     - Is it immutable or should anything be treated as immutable
     - Is it a utility class of static methods that should not be instantiated
 
+### Annotations
+
+- On methods:
+    - Method should be marked as `@Nullable` if they can return null values
+    - Method parameters should be marked as `@Nullable`.
+- On fields:
+    - Fields should often be marked as `@Nullable` if they are nullable. If the class
+    only exposes the field through a simple getter method, using the annotation on field is optional,
+    but on the getter method it is required.
```

Comment:
IMO we should make `@Nullable` required for all fields that can be null. Because it makes the rule simpler and I don't think we gain much by having the exception for fields that are only used as simple properties.

But we can discuss that at a meeting or sometime tomorrow perhaps.